### ISSUE TITLE
Auto model & pipeline for image-text-to-image-text models

### ISSUE THREAD

**Issue by leloykun** [huggingface/transformers#32926]
### Feature request

This is a tracker issue for work on _interleaved_ in-and-out image-text generation.

There are now >= 4 open-source models that can do _interleaved_ image-text generation--and many more are expected to be released. Thus, it would now be practical & useful for us to (1) add native support for such models and (2) standardize the logic flow of data through processors and pipelines as done in https://github.com/huggingface/transformers/issues/31911 and https://github.com/huggingface/transformers/pull/32472

Model | Github | Notes | PR
--- | --- | --- | ---
[Chameleon](https://arxiv.org/abs/2405.09818) | https://github.com/facebookresearch/chameleon | - | https://github.com/huggingface/transformers/pull/32013
[Anole](https://arxiv.org/abs/2407.06135) | https://github.com/GAIR-NLP/anole | - | https://github.com/huggingface/transformers/pull/32013
[Lumina-mGPT](https://arxiv.org/abs/2408.02657) | https://github.com/Alpha-VLLM/Lumina-mGPT | - | -
[Transfusion](https://arxiv.org/abs/2408.11039) | - | - | -
[XGen-MM](https://arxiv.org/abs/2408.08872) | https://github.com/salesforce/LAVIS/tree/xgen-mm | The paper & the github repo don't actually demonstrate interleaved image-text generation yet, but they did train the model on such datasets & the model architecture(s) is perfectly suited for it | -

Initial work for Chameleon & Anole can be found here: https://github.com/huggingface/transformers/pull/32013 for reference.

TODOs:
- [ ] Add support for interleaved image-text generation with:
  - [ ] Chameleon -> https://github.com/huggingface/transformers/pull/32013
  - [ ] Anole -> https://github.com/huggingface/transformers/pull/32013
  - [ ] Lumina-mGPT
  - [ ] Transfusion
  - [ ] XGen-MM
- [ ] Add auto model for image-text-to-image-text
  - [ ] [Optional] Add auto model for image-to-image-text
  - [ ] [Optional] Add auto model for text-to-image-text
- [ ] Add pipeline for image-text-to-image-text
  - [ ] [Optional] Add pipeline for image-to-image-text
  - [ ] [Optional] Add pipeline for text-to-image-text

### Motivation

.

### Your contribution

I've already started work on Chameleon & Anole here: https://github.com/huggingface/transformers/pull/32013

But I'm currently blocked by (1) not having enough time due to other responsibilities and (2) not having enough compute resources.

Any help would be appreciated!

**Comment by zucchini-nlp** [comment]
FYI @NielsRogge and @merveenoyan , you've been discussing recently tags for these kinds of models on the hub

**Comment by GargDivanshu** [comment]
@leloykun saw your comment on issue #33905 (Implement LlamaGen for Image Generation) 

Want to work on these issues, can you tell where to begin with ? 
I am reading #31911 as you mentioned above

**Comment by leloykun** [comment]
@GargDivanshu You might also wanna take a look at #32013

You can start by adding some of the missing tests and such to gain familiarity with the code there. And once you're ready, I can help you implement multimodal in-and-out for the other models.

**Comment by GargDivanshu** [comment]
perfect, moving to #32013

**Comment by merveenoyan** [comment]
@zucchini-nlp I think this falls under `any-to-any` in Hub side but not sure if in transformers we should have a separate pipeline given we don't have a ton of these models as of now and given the shift to `any-to-any` we will have to have another pipeline for models that can take audio input or output on top of the modalities here. @NielsRogge

**Comment by zucchini-nlp** [comment]
Yes, I agree it should be any-to-any. Was just adding you in the loop since some contributors are working on adding these types of models :)
